salt/modules/mac_brew_pkg.py

```diff
@@ -138,6 +147,52 @@ def _list_pkgs_from_context(versions_as_list):
         return ret
 
 
+def homebrew_prefix(raise_err=True):
+    """
+    Returns the full path to the homebrew prefix
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.homebrew_prefix
+    """
+
+    # Try HOMEBREW_PREFIX env variable
+    env_homebrew_prefix = "HOMEBREW_PREFIX"
+    if env_homebrew_prefix in os.environ:
+        log.debug(f"{env_homebrew_prefix} is set. Using it for homebrew prefix.")
+        return os.environ[env_homebrew_prefix]
+    else:
+        log.warning(f"{env_homebrew_prefix} is not set. Please, consider adding it.")
+
+    # Try brew --prefix
+    try:
+        log.debug("Trying to find homebrew prefix by running 'brew --prefix'")
+        import salt.modules.cmdmod
+
+        ret = salt.modules.cmdmod.run(
```

Comment:
I use `salt.modules.cmdmod.run` instead of `__salt__["cmd.run"]` because it is not available to use from `salt/utils/rsax931.py`

Reply:
With this change, now you can run `salt-*` commands with `sudo` and without the `-E` argument:

```sh
sudo salt-call test.version
[WARNING ] HOMEBREW_PREFIX is not set. Please, consider adding it.
local:
    3006.1
```

Before this change, you get:

```
Traceback (most recent call last):
  File "/opt/homebrew/bin/salt-call", line 8, in <module>
    sys.exit(salt_call())
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/scripts.py", line 438, in salt_call
    import salt.cli.call
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/cli/call.py", line 3, in <module>
    import salt.cli.caller
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/cli/caller.py", line 12, in <module>
    import salt.channel.client
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/channel/client.py", line 13, in <module>
    import salt.crypt
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/crypt.py", line 31, in <module>
    import salt.utils.rsax931
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/utils/rsax931.py", line 158, in <module>
    libcrypto = _init_libcrypto()
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/utils/rsax931.py", line 111, in _init_libcrypto
    libcrypto = _load_libcrypto()
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/utils/rsax931.py", line 104, in _load_libcrypto
    return cdll.LoadLibrary(_find_libcrypto())
  File "/opt/homebrew/Cellar/salt/3006.1_1/libexec/lib/python3.10/site-packages/salt/utils/rsax931.py", line 96, in _find_libcrypto
    raise OSError("Cannot locate OpenSSL libcrypto")
OSError: Cannot locate OpenSSL libcrypto
```